src/nodetool/nodes/kie/image.py

```diff
@@ -0,0 +1,502 @@
+"""Kie.ai image generation nodes.
+
+This module provides nodes for generating images using Kie.ai's various APIs:
+- 4O Image API (GPT-4o powered image generation)
+- Seedream (ByteDance's image generation model)
+- Z-Image (High-quality image generation)
+- Nano Banana (Google's efficient image model)
+- Flux Pro (Black Forest Labs text-to-image)
+"""
+
+import asyncio
+from abc import abstractmethod
+from enum import Enum
+from typing import Any, ClassVar
+
+import aiohttp
+from pydantic import Field
+
+from nodetool.config.logging_config import get_logger
+from nodetool.metadata.types import ImageRef
+from nodetool.workflows.base_node import BaseNode
+from nodetool.workflows.processing_context import ProcessingContext
+
+log = get_logger(__name__)
+
+# Kie.ai API base URL
+KIE_API_BASE_URL = "https://api.kie.ai"
+
+
+class KieBaseNode(BaseNode):
+    """Base class for Kie.ai API nodes with polling logic for task completion.
+
+    kie, ai, image generation, api
+
+    This base class encapsulates the common pattern for Kie.ai APIs:
+    1. Submit a task (POST request)
+    2. Poll for task completion (GET request)
+    3. Download the result
+    """
+
+    # Polling configuration
+    poll_interval: float = Field(
+        default=2.0,
+        description="Interval in seconds between status checks.",
+        ge=0.5,
+        le=30.0,
+    )
+    max_poll_attempts: int = Field(
+        default=120,
+        description="Maximum number of polling attempts before timeout.",
+        ge=1,
+        le=600,
+    )
+
+    @classmethod
+    def is_visible(cls) -> bool:
+        return cls is not KieBaseNode
+
+    async def _get_api_key(self, context: ProcessingContext) -> str:
+        """Get the Kie.ai API key from secrets."""
+        api_key = await context.get_secret("KIE_API_KEY")
+        if not api_key:
+            raise ValueError(
+                "KIE_API_KEY secret is not configured. "
+                "Please set the KIE_API_KEY secret in your configuration."
+            )
+        return api_key
+
+    def _get_headers(self, api_key: str) -> dict[str, str]:
+        """Get common headers for Kie.ai API requests."""
+        return {
+            "Authorization": f"Bearer {api_key}",
+            "Content-Type": "application/json",
+        }
+
+    @abstractmethod
+    def _get_submit_endpoint(self) -> str:
```

Comment:
the class should only implement one base endpoint function, other endpoints should be derived from that